proto/engine/v1/json_marshal_unmarshal.go

```diff
@@ -1,47 +1,215 @@
+)
+
+// HexBytes implements a custom json.Marshaler/Unmarshaler for byte slices that encodes them as
+// hex strings per the Ethereum JSON-RPC specification.
+type HexBytes []byte
+
+// Quantity implements a custom json.Marshaler/Unmarshaler for uint64 that encodes them as
+// big-endian hex strings per the Ethereum JSON-RPC specification.
+type Quantity uint64
+
+func (b HexBytes) MarshalJSON() ([]byte, error) {
+	return json.Marshal(hexutil.Encode(b))
+}
+
+func (b *HexBytes) UnmarshalJSON(enc []byte) error {
+	if len(enc) == 0 {
+		*b = make([]byte, 0)
+		return nil
+	}
+	var hexString string
+	if err := json.Unmarshal(enc, &hexString); err != nil {
+		return err
+	}
+	dst, err := hexutil.Decode(hexString)
+	if err != nil {
+		return err
+	}
+	*b = dst
+	return nil
+}
+
+func (q Quantity) MarshalJSON() ([]byte, error) {
+	enc := make([]byte, 8)
+	binary.BigEndian.PutUint64(enc, uint64(q))
+	return json.Marshal(hexutil.Encode(enc))
+}
+
+func (q *Quantity) UnmarshalJSON(enc []byte) error {
+	if len(enc) == 0 {
+		*q = 0
+		return nil
+	}
+	var hexString string
+	if err := json.Unmarshal(enc, &hexString); err != nil {
+		return err
+	}
+	dst, err := hexutil.Decode(hexString)
+	if err != nil {
+		return err
+	}
+	*q = Quantity(binary.BigEndian.Uint64(dst))
+	return nil
+}
+
+type executionPayloadJSON struct {
+	ParentHash    HexBytes   `json:"parentHash"`
+	FeeRecipient  HexBytes   `json:"feeRecipient"`
+	StateRoot     HexBytes   `json:"stateRoot"`
+	ReceiptsRoot  HexBytes   `json:"receiptsRoot"`
+	LogsBloom     HexBytes   `json:"logsBloom"`
+	Random        HexBytes   `json:"random"`
+	BlockNumber   Quantity   `json:"blockNumber"`
+	GasLimit      Quantity   `json:"gasLimit"`
+	GasUsed       Quantity   `json:"gasUsed"`
+	Timestamp     Quantity   `json:"timestamp"`
+	ExtraData     HexBytes   `json:"extraData"`
+	BaseFeePerGas HexBytes   `json:"baseFeePerGas"`
+	BlockHash     HexBytes   `json:"blockHash"`
+	Transactions  []HexBytes `json:"transactions"`
+}
 
 // MarshalJSON defines a custom json.Marshaler interface implementation
-// that uses protojson underneath the hood, as protojson will respect
-// proper struct tag naming conventions required for the JSON-RPC engine API to work.
+// that uses custom json.Marshalers for the HexBytes and Quantity types.
 func (e *ExecutionPayload) MarshalJSON() ([]byte, error) {
-	return protojson.Marshal(e)
+	transactions := make([]HexBytes, len(e.Transactions))
+	for i, tx := range e.Transactions {
+		transactions[i] = tx
+	}
+	return json.Marshal(executionPayloadJSON{
+		ParentHash:    e.ParentHash,
+		FeeRecipient:  e.FeeRecipient,
+		StateRoot:     e.StateRoot,
+		ReceiptsRoot:  e.ReceiptsRoot,
+		LogsBloom:     e.LogsBloom,
+		Random:        e.Random,
+		BlockNumber:   Quantity(e.BlockNumber),
+		GasLimit:      Quantity(e.GasLimit),
+		GasUsed:       Quantity(e.GasUsed),
+		Timestamp:     Quantity(e.Timestamp),
+		ExtraData:     e.ExtraData,
+		BaseFeePerGas: e.ExtraData,
```

Comment:
Surprised tests didn't fail because of this. Perhaps add more entropy to the test?

```suggestion
		BaseFeePerGas: e.BaseFeePerGas,
```